ambry-quota/src/main/java/com/github/ambry/quota/storage/StorageQuotaEnforcer.java

```diff
@@ -267,16 +232,19 @@ private void initMap(Map<String, Map<String, Long>> mapWithValue, Map<String, Ma
     long quotaValue = -1L;
     long currentUsage = 0L;
     try {
-      short accountId = RestUtils.getAccountFromArgs(restRequest.getArgs()).getId();
-      short containerId = RestUtils.getContainerFromArgs(restRequest.getArgs()).getId();
-      quotaValue = storageQuota.computeIfAbsent(String.valueOf(accountId), k -> new HashMap<>())
-          .getOrDefault(String.valueOf(containerId), -1L);
+      Container container = RestUtils.getContainerFromArgs(restRequest.getArgs());
```

Comment:
Nit: `getQuotaAndUsage` and `charge` methods have these few lines in common. Maybe you can create a private method `getQuotaValueForContainer` to fetch quotavalue for a container, and use it in both methods.